diff --git a/sites/all/modules/contrib/field_slideshow/field_slideshow.module b/sites/all/modules/contrib/field_slideshow/field_slideshow.module
index 687d3b92a..c0f1e52b4 100644
--- a/sites/all/modules/contrib/field_slideshow/field_slideshow.module
+++ b/sites/all/modules/contrib/field_slideshow/field_slideshow.module
@@ -683,11 +683,17 @@ function field_slideshow_field_formatter_view($entity_type, $entity, $field, $in
   if ($settings['slideshow_caption'] != '') {
     foreach ($items as $delta => $item) {
       if ($field['type'] == 'media' || $field['type'] == 'field_collection') {
-        if ($field['type'] == 'media') $items[$delta]['caption'] = $items[$delta]['file']->{$settings}['slideshow_caption'];
-        elseif ($field['type'] == 'field_collection') $items[$delta]['caption'] = $items[$delta][$settings['slideshow_caption']];
-        if (!empty($items[$delta]['caption']) && isset($items[$delta]['caption'][$langcode]))
-          $items[$delta]['caption'] = filter_xss($items[$delta]['caption'][$langcode][0]['value']);
-        else $items[$delta]['caption'] = '';
+      if ($field['type'] == 'media') $items[$delta]['caption'] = $items[$delta]['file']->$settings['slideshow_caption'];
+      elseif ($field['type'] == 'field_collection'){
+        $items[$delta]['caption'] = $items[$delta][$settings['slideshow_caption']];
+      }
+
+      if(!empty($items[$delta]['caption']['und'][0]['format']) && $items[$delta]['caption']['und'][0]['format'] != 'plain_text'){
+        $items[$delta]['caption'] = $items[$delta]['caption'][$langcode][0]['value'];
+      }
+      elseif (!empty($items[$delta]['caption']) && isset($items[$delta]['caption'][$langcode]))
+        $items[$delta]['caption'] = filter_xss($items[$delta]['caption'][$langcode][0]['value']);
+      else $items[$delta]['caption'] = '';
       }
       else $items[$delta]['caption'] = filter_xss($item[$settings['slideshow_caption']]);
     }
