From 9508e8a10c82f10cf5d9a5b1f183daea7d21ed3c Mon Sep 17 00:00:00 2001
From: Matthew Grasmick <matthew.grasmick@acquia.com>
Date: Mon, 27 Jan 2014 16:55:49 -0500
Subject: [PATCH] Issue #1366226: Allow long text fields as captions.

---
 field_slideshow.module | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/field_slideshow.module b/field_slideshow.module
index fc42801..5b6029d 100644
--- a/field_slideshow.module
+++ b/field_slideshow.module
@@ -162,7 +162,7 @@ function field_slideshow_field_formatter_settings_form($field, $instance, $view_
   }
 
   if ($field['type'] == 'media' || $field['type'] == 'field_collection')
-    $captions = _field_slideshow_get_fields(array('text'), $field['type'], $field['field_name']);
+    $captions = _field_slideshow_get_fields(array('text', 'text_long', 'text_with_summary'), $field['type'], $field['field_name']);
   else $captions = array(
     'title'   => t('Title text'),
     'alt'     => t('Alt text'),
-- 
1.8.0

