diff --git a/sites/all/modules/contrib/field_slideshow/field_slideshow.module b/sites/all/modules/contrib/field_slideshow/field_slideshow.module
index 29f2e547..a0b548d2 100644
--- a/sites/all/modules/contrib/field_slideshow/field_slideshow.module
+++ b/sites/all/modules/contrib/field_slideshow/field_slideshow.module
@@ -648,7 +648,7 @@ function field_slideshow_field_formatter_view($entity_type, $entity, $field, $in
   if ($settings['slideshow_caption'] != '') {
     foreach ($items as $delta => $item) {
       if ($field['type'] == 'media' || $field['type'] == 'field_collection') {
-        if ($field['type'] == 'media') $items[$delta]['caption'] = $items[$delta]['file']->$settings['slideshow_caption'];
+        if ($field['type'] == 'media') $items[$delta]['caption'] = $items[$delta]['file']->{$settings['slideshow_caption']};
         elseif ($field['type'] == 'field_collection') $items[$delta]['caption'] = $items[$delta][$settings['slideshow_caption']];
         if (!empty($items[$delta]['caption']) && isset($items[$delta]['caption'][$langcode]))
           $items[$delta]['caption'] = $items[$delta]['caption'][$langcode][0]['value'];
@@ -719,7 +719,7 @@ function field_slideshow_field_formatter_view($entity_type, $entity, $field, $in
           }
         break;
         case 'link_field':
-          if ($field['type'] == 'media') $pathfield = $item['file']->$settings[$setting];
+          if ($field['type'] == 'media') $pathfield = $item['file']->{$settings[$setting]};
           elseif ($field['type'] == 'field_collection') $pathfield = $item[$settings[$setting]];
           if (isset($pathfield[$langcode])) {
             $uri = array(
@@ -729,7 +729,7 @@ function field_slideshow_field_formatter_view($entity_type, $entity, $field, $in
           }
         break;
         case 'node_reference':
-          if ($field['type'] == 'media') $pathfield = $item['file']->$settings[$setting];
+          if ($field['type'] == 'media') $pathfield = $item['file']->{$settings[$setting]};
           elseif ($field['type'] == 'field_collection') $pathfield = $item[$settings[$setting]];
           if (isset($pathfield[$langcode])) {
             $uri = array(
@@ -739,7 +739,7 @@ function field_slideshow_field_formatter_view($entity_type, $entity, $field, $in
           }
         break;
         case 'entityreference':
-          if ($field['type'] == 'media') $pathfield = $item['file']->$settings[$setting];
+          if ($field['type'] == 'media') $pathfield = $item['file']->{$settings[$setting]};
           elseif ($field['type'] == 'field_collection') $pathfield = $item[$settings[$setting]];
           if (isset($pathfield[$langcode])) {
             $uri = array(
