diff --git a/sites/all/modules/contrib/media_youtube/includes/MediaInternetYouTubeHandler.inc b/sites/all/modules/contrib/media_youtube/includes/MediaInternetYouTubeHandler.inc
index b636215..0d4a4ee 100644
--- a/sites/all/modules/contrib/media_youtube/includes/MediaInternetYouTubeHandler.inc
+++ b/sites/all/modules/contrib/media_youtube/includes/MediaInternetYouTubeHandler.inc
@@ -111,15 +111,25 @@ class MediaInternetYouTubeHandler extends MediaInternetBaseHandler {
     $uri = file_stream_wrapper_uri_normalize('youtube://' . $type . '/' . check_plain($id));
     $external_url = file_create_url($uri);
     $oembed_url = url('https://www.youtube.com/oembed', array('query' => array('url' => $external_url, 'format' => 'json')));
-    $response = drupal_http_request($oembed_url, array('method' => 'HEAD'));
+    $response = drupal_http_request($oembed_url);

-    if ($response->code == 401) {
-      throw new MediaInternetValidationException('Embedding has been disabled for this YouTube video.');
+    if (!isset($response->error)) {
+      $data = drupal_json_decode($response->data);
+      if (!empty($data)) {
+        return TRUE;
+      }
+      else {
+        $error_data = t('Unspecific');
+        if (is_string($response->data)) {
+          $error_data = $response->data;
+        }
+        throw new MediaInternetValidationException("The YouTube video ID is invalid, video was deleted or is disabled for embedding. Error: {$error_data}");
+        return;
+      }
     }
-    elseif ($response->code != 200) {
-      throw new MediaInternetValidationException('The YouTube video ID is invalid or the video was deleted.');
+    else {
+      throw new Exception("Error Processing Request. (Error: {$response->code}, {$response->error})");
+      return;
     }
-
-    return TRUE;
   }
 }
