diff --git a/sites/all/modules/contrib/webform/css/webform.css b/sites/all/modules/contrib/webform/css/webform.css
index 956209f..b8e16cc 100644
--- a/sites/all/modules/contrib/webform/css/webform.css
+++ b/sites/all/modules/contrib/webform/css/webform.css
@@ -34,6 +34,38 @@ html.js input.webform-calendar {
 .webform-container-inline.webform-component-textarea .form-textarea-wrapper {
   display: inline-block;
 }
+
+/* Reset so that these appear the same as the label elements they replace. */
+fieldset.fieldset-invisible,
+fieldset.fieldset-invisible > legend {
+  margin: 0;
+  padding: 0;
+  border: none;
+  border-radius: 0;
+  background: inherit;
+  position: static;
+  color: inherit;
+  height: auto;
+  width: auto;
+  font-family: inherit;
+  text-indent: 0;
+  line-height: inherit;
+  text-shadow: unset;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+}
+fieldset.fieldset-invisible > legend {
+  font-weight: bold;
+  font-size: 0.929em;
+}
+/* This margin causes the fieldset to be too big. */
+fieldset.fieldset-invisible > div > div.form-item:last-child,
+fieldset.fieldset-invisible > table {
+  margin-bottom: 0;
+}
+
 .webform-component-textarea .grippie {
   display: block;
 }
diff --git a/sites/all/modules/contrib/webform/includes/webform.admin.inc b/sites/all/modules/contrib/webform/includes/webform.admin.inc
index aca45fa..a2a2559 100644
--- a/sites/all/modules/contrib/webform/includes/webform.admin.inc
+++ b/sites/all/modules/contrib/webform/includes/webform.admin.inc
@@ -255,6 +255,13 @@ function webform_admin_settings() {
     '#description' => t('When mapping emails addresses to a select component, limit the choice to components with less than the amount of options indicated. This is to avoid flooding the email settings form.'),
   );
 
+  $form['advanced']['webform_fieldset_wrap'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Use fieldsets for grouped components'),
+    '#default_value' => webform_variable_get('webform_fieldset_wrap'),
+    '#description' => t('Components containing multiple fields, such as checkboxes and radio buttons, will be wrapped in fieldsets. This improves the accessibility of webforms and helps them conform to web standards, but may require updates to the CSS of the active theme.'),
+  );
+
   $form = system_settings_form($form);
   $form['#theme'] = 'webform_admin_settings';
   array_unshift($form['#submit'], 'webform_admin_settings_submit');
diff --git a/sites/all/modules/contrib/webform/webform.install b/sites/all/modules/contrib/webform/webform.install
index e137831..664f275 100644
--- a/sites/all/modules/contrib/webform/webform.install
+++ b/sites/all/modules/contrib/webform/webform.install
@@ -796,6 +796,7 @@ function webform_uninstall() {
   variable_del('webform_default_format');
   variable_del('webform_format_override');
   variable_del('webform_email_select_max');
+  variable_del('webform_fieldset_wrap');
   variable_del('webform_node_types_primary');
   variable_del('webform_date_type');
   variable_del('webform_export_format');
diff --git a/sites/all/modules/contrib/webform/webform.module b/sites/all/modules/contrib/webform/webform.module
index 0786999..a7c9257 100644
--- a/sites/all/modules/contrib/webform/webform.module
+++ b/sites/all/modules/contrib/webform/webform.module
@@ -3754,6 +3754,23 @@ function theme_webform_element($variables) {
     $variables['element']['#id'] = NULL;
   }
 
+  // Determine whether or not this element has form control children. If so and
+  // if webform_fieldset_wrap is TRUE, wrap them in a fieldset and use legend
+  // instead of label.
+  $has_element_children = FALSE;
+  if (webform_variable_get('webform_fieldset_wrap')) {
+    foreach (array_keys($element) as $key) {
+      if (substr($key, 0, 1) !== '#') {
+        $has_element_children = TRUE;
+        break;
+      }
+    }
+  }
+
+  if ($has_element_children) {
+    $output .= '<fieldset class="fieldset-invisible">';
+  }
+
   switch ($element['#title_display']) {
     case 'inline':
       $output .= $description[$above];
@@ -3761,7 +3778,18 @@ function theme_webform_element($variables) {
     case 'before':
     case 'invisible':
     case 'after':
-      $title = ' ' . theme('form_element_label', $variables);
+      if ($has_element_children) {
+        $title = '<legend>' . $element['#title'];
+
+        if ($element['#required']) {
+          $title .= ' ' . theme('form_required_marker', $variables);
+        }
+
+        $title .= '</legend>';
+      }
+      else {
+        $title = ' ' . theme('form_element_label', $variables);
+      }
       break;
   }
 
@@ -3788,6 +3816,11 @@ function theme_webform_element($variables) {
   $output .= "\n";
 
   $output .= $description[!$above];
+
+  if ($has_element_children) {
+    $output .= '</fieldset>';
+  }
+
   $output .= "</div>\n";
 
   return $output;
@@ -4414,6 +4447,10 @@ function webform_variable_get($variable) {
     case 'webform_disabled_components':
       $result = variable_get('webform_disabled_components', array());
       break;
+
+    case 'webform_fieldset_wrap':
+      $result = variable_get('webform_fieldset_wrap', FALSE);
+      break;
   }
   return $result;
 }
